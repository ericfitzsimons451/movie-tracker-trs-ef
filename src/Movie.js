import React from 'react'

const Movie = (movie) => {
    console.log('movie', movie)

    return (
        <div>
            <h1>{movie.movie.title}</h1>
        </div>
    )
}

export default Movie