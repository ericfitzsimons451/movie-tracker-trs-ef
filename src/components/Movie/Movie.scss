.movie-card {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    z-index: 10;
    background: white;
    width: 300px;
    min-height: 300px;
    border: 1px solid black;
    padding: 10px;
    margin: 5px;
    border-radius: 10px;
    img {
        width: 80%;
    }
    button {
        background-color: blueviolet;
    }
}

.movie-title {
    font-size: 2rem;
}




