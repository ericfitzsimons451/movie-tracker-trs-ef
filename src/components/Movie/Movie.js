import React from 'react'
import './Movie.scss'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

const Movie = ({ name, poster_path, release_date, vote_average, overview }) => {
    return (
        <div className='movie-card'>
            <h3>{name}</h3>
            <img src={`http://image.tmdb.org/t/p/original/${poster_path}`} alt="poster image" />
            <h3>Released: {release_date}</h3>
            <h3>Vote Average: {vote_average}</h3>
            <p>Overview: {overview}</p>
            <button>Add To Favorites</button>
        </div>
    )
}

Movie.propTypes = {
    movie: PropTypes.object
}

const mapStateToProps = (state) => ({
    movies: state.movies
})

export default connect(mapStateToProps)(Movie)