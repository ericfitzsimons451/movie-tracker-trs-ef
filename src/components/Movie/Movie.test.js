import Movie from "../Movie/Movie";
import React from "react";
import { shallow } from "enzyme";
import { mapStateToProps } from './Movie';

describe.skip("Movie", () => {
  let wrapper;
  it('should match the snapshot with all data passed down', () => {
    const mockProps = {
      name: "How to Train Your Dragon: The Hidden World",
      id: 166428,
      poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
      release_date: "2019-01-03",
      vote_average: 7.8,
      overview:
        "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind."
    },
    wrapper = shallow(<Movie movie={mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  // describe("mapStateToProps", () => {
  //   it("should update props from the Store with a user", () => {
  //     const mockUser = {
  //       email: "tman2272@aol.com",
  //       id: 1,
  //       name: "Taylor"
  //     };
  //     const mockState = {
  //       user: mockUser
  //     };
  //     const expectedProps = {
  //       user: {
  //         email: "tman2272@aol.com",
  //         id: 1,
  //         name: "Taylor"
  //       }
  //     };
  //     const mappedProps = mapStateToProps(mockState);
  //     expect(mappedProps).toEqual(expectedProps);
  //   });
  // });

  // describe('addFavorite', () => {})
});
