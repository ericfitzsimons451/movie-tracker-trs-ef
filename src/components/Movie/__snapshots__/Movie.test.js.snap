// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot with all data passed down 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
